### PR TITLE
wp-admin/js/post.js documentation

### LINKED ISSUE #2: Document wp-admin/js/post.js

### PR DESCRIPTION
**Pull request by moorscode**
By Jip and Bob.

Fixes #2

### CODE REVIEW

File: src/wp-admin/js/post.js [hunk 1]
```diff
@@ -2,172 +2,244 @@
...
-				$( '#commentsdiv .spinner' ).removeClass( 'is-active' );
-
-				if ( 'object' == typeof r && r.responses[0] ) {
-					$('#the-comment-list').append( r.responses[0].data );
+	/**
+	 * Control loading of comments on the post and term edit pages
+	 *
+	 * @type {{st: number, get: commentsBox.get, load: commentsBox.load}}
+	 *
+	 * @namespace commentsbox
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: I think it's wise to keep the namespace name in line with the actual global variable, if people copy and paste the namespace name to their code it just works.

File: src/wp-admin/js/post.js [hunk 2]
```diff
@@ -2,172 +2,244 @@
...
+	 */
+	commentsBox = {
+		// Comment offset to use when fetching new comments.
+		st : 0,
+
+		/**
+		 * Fetch comments using AJAX and display them in the box
+		 *
+		 * @param {int} total Total number of comments for this post.
+		 * @param {int} num   Optional. Number of comments to fetch, defaults to 20.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: Optional comment should be document using brackets, like this: `@param {int} [num]`

File: src/wp-admin/js/post.js [hunk 3]
```diff
@@ -450,34 +538,59 @@ jQuery(document).ready( function($) {
...
 
+		// After submitting a new taxonomy, re-focus the input field.
+		$('#' + taxonomy + '-add-submit').click( function() {
+			$('#new' + taxonomy).focus();
+		});
+
+		/**
+		 * Before adding a new taxonomy, disable submit button.
+		 *
+		 * @param {object} s Taxonomy object which will be added.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: `Object` should be with a capital letter, see: http://usejsdoc.org/tags-type.html

File: src/wp-admin/js/post.js [hunk 4]
```diff
@@ -450,34 +538,59 @@ jQuery(document).ready( function($) {
...
+		 * @returns {*}
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: Isn't this always guaranteed to be an object, given that `s` is an object?

File: src/wp-admin/js/post.js [hunk 5]
```diff
@@ -450,34 +538,59 @@ jQuery(document).ready( function($) {
...
 			return s;
 		};
 
+		/**
+		 * After a taxonomy has been added.
+		 *
+		 * Re-enable submit button.
+		 * If the taxonomy has a parent place the taxonomy underneath the parent.
+		 *
+		 * @param {object} r Response.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: ^ Object

File: src/wp-admin/js/post.js [hunk 6]
```diff
@@ -450,34 +538,59 @@ jQuery(document).ready( function($) {
...
 				return false;
+			}
+
 			s.data += '&' + $( ':checked', '#'+taxonomy+'checklist' ).serialize();
 			$( '#' + taxonomy + '-add-submit' ).prop( 'disabled', true );
 			return s;
 		};
 
+		/**
+		 * After a taxonomy has been added.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: This is not a sentence, maybe have the title be: `Re-enable submit button after a taxonomy has been added.`

File: src/wp-admin/js/post.js (hunk 5 above)
Comment (atimmer) [inline src/wp-admin/js/post.js]: Maybe document what response this is. As far as I can see it's a response from an AJAX request originated in wpList.

File: src/wp-admin/js/post.js [hunk 7]
```diff
@@ -510,36 +625,62 @@ jQuery(document).ready( function($) {
...
-			$('table#list-table').show();
-		}, addBefore: function( s ) {
-			s.data += '&post_id=' + $('#post_ID').val();
-			return s;
-		}
+		$( '#the-list' ).wpList( {
+			/**
+			 * Add current post_ID to request to fetch custom fields
+			 *
+			 * @param {object} s Request object.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: ^ Object

File: src/wp-admin/js/post.js [hunk 8]
```diff
@@ -510,36 +625,62 @@ jQuery(document).ready( function($) {
...
+	 * Publish Post box (#submitdiv)
+	 */
 	if ( $('#submitdiv').length ) {
 		stamp = $('#timestamp').html();
 		visibility = $('#post-visibility-display').html();
 
+		/**
+		 * When the visibility of a post changes sub-options should be shown or hidden.
+		 *
+		 * @returns void
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: We've decided not to document `void` return types.

File: src/wp-admin/js/post.js [hunk 9]
```diff
@@ -570,7 +713,10 @@ jQuery(document).ready( function($) {
 				publishOn = postL10n.publishOnPast;
 				$('#publish').val( postL10n.update );
 			}
-			if ( originalDate.toUTCString() == attemptedDate.toUTCString() ) { //hack
+
+			// If the date is the same, just force it to itself for update triggers.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: This is a not so readable sentence part: `force it to itself for`?

File: src/wp-admin/js/post.js [hunk 10]
```diff
@@ -570,7 +713,10 @@ jQuery(document).ready( function($) {
...
+			if ( originalDate.toUTCString() == attemptedDate.toUTCString() ) {
+				// Set to what it is already.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: `already is`?

File: src/wp-admin/js/post.js [hunk 11]
```diff
@@ -717,21 +877,29 @@ jQuery(document).ready( function($) {
...
-	} // end submitdiv
+	}
 
-	// permalink
+	/**
+	 * Permalink aka slug aka post_name editing
+	 *
+	 * @global
+	 *
+	 * @returns void
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: ^

File: src/wp-admin/js/post.js [hunk 12]
```diff
@@ -717,21 +877,29 @@ jQuery(document).ready( function($) {
...
 			$('#post_status').val( $('#hidden_post_status').val() );
 			updateText();
 			event.preventDefault();
 		});
-	} // end submitdiv
+	}
 
-	// permalink
+	/**
+	 * Permalink aka slug aka post_name editing
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: I think this function needs so more explanation. As far as I can see it is initializing the permalink section, but that could be documented in the JSDoc.

File: src/wp-admin/js/post.js [hunk 13]
```diff
@@ -818,6 +998,15 @@ jQuery(document).ready( function($) {
 		editPermalink();
 	});
 
+	/**
+	 * Title screen reader text handler.
+	 *
+	 * @param {string} id Optional. HTML ID to add the screen reader helper text to.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: Same as above: `[id]` should be used here.

File: src/wp-admin/js/post.js [hunk 14]
```diff
@@ -818,6 +998,15 @@ jQuery(document).ready( function($) {
 		editPermalink();
 	});
 
+	/**
+	 * Title screen reader text handler.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: I really need more explanation about what this function actually does.

File: src/wp-admin/js/post.js [hunk 15]
```diff
@@ -844,19 +1033,24 @@ jQuery(document).ready( function($) {
 
 	wptitlehint();
 
-	// Resize the visual and text editors
+	// Resize the wysiwyg and plain text editors.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: `WYSIWYG` is spelled with capital letters.

File: src/wp-admin/js/post.js [hunk 16]
```diff
@@ -844,19 +1033,24 @@ jQuery(document).ready( function($) {
...
-			// Hide the resize handle
+			// Hide the resize handle.
 			$('#content-resize-handle').hide();
 			return;
 		}
 
+		/**
+		 * Handle drag event.
+		 *
+		 * @param {object} event Event containing details about the drag.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: ^ `Object`

File: src/wp-admin/js/post.js [hunk 17]
```diff
@@ -894,7 +1091,7 @@ jQuery(document).ready( function($) {
 
 			$document.off( '.wp-editor-resize' );
 
-			// sanity check
+			// Normalize height to stay within acceptable ranges.
```
Comment (atimmer) [inline src/wp-admin/js/post.js]: Maybe prefix with `Sanity check:` to make it clear that this is the sanity check the JSDoc for the function is talking about.

**Comment by atimmer** [comment]
CR Done 👷🏻

File: src/wp-admin/js/post.js [hunk 18]
```diff
@@ -1,173 +1,251 @@
...
-		$.post(ajaxurl, data,
-			function(r) {
-				r = wpAjax.parseAjaxResponse(r);
-				$('#commentsdiv .widefat').show();
-				$( '#commentsdiv .spinner' ).removeClass( 'is-active' );
-
-				if ( 'object' == typeof r && r.responses[0] ) {
-					$('#the-comment-list').append( r.responses[0].data );
+	/**
+	 * Control loading of comments on the post and term edit pages
```
Comment (terw-dan) [inline src/wp-admin/js/post.js]: Please add a period.

File: src/wp-admin/js/post.js [hunk 19]
```diff
@@ -1,173 +1,251 @@
...
+	 * @type {{st: number, get: commentsBox.get, load: commentsBox.load}}
+	 *
+	 * @namespace commentsBox
+	 */
+	commentsBox = {
+		// Comment offset to use when fetching new comments.
+		st : 0,
+
+		/**
+		 * Fetch comments using AJAX and display them in the box
```
Comment (terw-dan) [inline src/wp-admin/js/post.js]: Please add a period.

**Comment by terw-dan** [comment]
CR:
currently `@return` is used but we use `@returns` in other files. For consistency we should use `@returns`.

**Comment by terw-dan** [comment]
CR :+1:

**Comment by moorscode** [comment]
Created the ticket https://core.trac.wordpress.org/ticket/38118 @boblinthorst

**Pull request closed without merging**